libutils/string_lib.c

```diff
@@ -62,7 +62,7 @@ size_t StringCopy(const char *const from, char *const to, const size_t buf_size)
     assert(from != NULL);
     assert(to != NULL);
     assert(from != to);
-    assert(buf_size >= 0);
+    assert((signed)buf_size >= 0);
```

Comment:
`buf_size` is `size_t`. There should be no problem comparing that to `0`. This looks wrong.

Reply:
I have reverted the changes. 

These changes were required if we build the package over gcc but we are using clang in travis for **“env: JOB_TYPE=compile_and_unit_test COVERAGE=no”** so these changes are not required.

Reply:
You have not reverted the change, it's still here. I think you're doing this because of a warning for comparison always true, because of the sign of the type.

We typically disable this by adding `-Wno-sign-compare`.

Reply:
Yes, my intention for change was the same as you described.
I have reverted that type-conversion now, as that is not required with clang being used in our travis-ci build.